internal/services/batch/batch_pool.go

```diff
@@ -373,6 +373,46 @@ func flattenBatchPoolIdentityReferenceToIdentityID(ref *batch.ComputeNodeIdentit
 	return ""
 }
 
+func flattenBatchPoolUserAccount(d *pluginsdk.ResourceData, account *batch.UserAccount) map[string]interface{} {
+	userAccount := make(map[string]interface{})
+	userAccount["name"] = *account.Name
+	userAccount["elevation_level"] = string(account.ElevationLevel)
+	userAccountIndex := -1
+	if num, ok := d.GetOk("user_accounts.#"); ok {
+		n := num.(int)
+		for i := 0; i < n; i++ {
+			if src, nameOk := d.GetOk(fmt.Sprintf("user_accounts.%d.name", i)); nameOk && src == *account.Name {
+				userAccount["password"] = d.Get(fmt.Sprintf("user_accounts.%d.password", i)).(string)
+				userAccountIndex = i
+				break
+			}
+		}
+	}
+	if account.LinuxUserConfiguration != nil {
+		linuxUserConfig := make(map[string]interface{})
+		if account.LinuxUserConfiguration.UID != nil {
+			linuxUserConfig["uid"] = *account.LinuxUserConfiguration.UID
+			linuxUserConfig["gid"] = *account.LinuxUserConfiguration.Gid
+		}
+		if userAccountIndex > -1 {
+			if sshPrivateKey, ok := d.GetOk(fmt.Sprintf("user_accounts.%d.linux_user_configuration.0.ssh_private_key", userAccountIndex)); ok {
+				linuxUserConfig["ssh_private_key"] = sshPrivateKey
+			}
+		}
+		userAccount["linux_user_configuration"] = []interface{}{
+			linuxUserConfig,
+		}
+	}
+	if account.WindowsUserConfiguration != nil {
+		loginMode := make(map[string]interface{})
+		loginMode["login_mode"] = string(account.WindowsUserConfiguration.LoginMode)
+		userAccount["windows_user_configuration"] = []interface{}{
+			loginMode,
+		}
+	}
+	return userAccount
+}
```

Comment:
please use whitespace for readability
```suggestion
	userAccountIndex := -1
	
	if num, ok := d.GetOk("user_accounts.#"); ok {
		n := num.(int)
		for i := 0; i < n; i++ {
			if src, nameOk := d.GetOk(fmt.Sprintf("user_accounts.%d.name", i)); nameOk && src == *account.Name {
				userAccount["password"] = d.Get(fmt.Sprintf("user_accounts.%d.password", i)).(string)
				userAccountIndex = i
				break
			}
		}
	}
	
	if account.LinuxUserConfiguration != nil {
		linuxUserConfig := make(map[string]interface{})
		
		if account.LinuxUserConfiguration.UID != nil {
			linuxUserConfig["uid"] = *account.LinuxUserConfiguration.UID
			linuxUserConfig["gid"] = *account.LinuxUserConfiguration.Gid
		}
		
		if userAccountIndex > -1 {
			if sshPrivateKey, ok := d.GetOk(fmt.Sprintf("user_accounts.%d.linux_user_configuration.0.ssh_private_key", userAccountIndex)); ok {
				linuxUserConfig["ssh_private_key"] = sshPrivateKey
			}
		}
		
		userAccount["linux_user_configuration"] = []interface{}{
			linuxUserConfig,
		}
	}
	
	if account.WindowsUserConfiguration != nil {
		loginMode := make(map[string]interface{})
		loginMode["login_mode"] = string(account.WindowsUserConfiguration.LoginMode)
		userAccount["windows_user_configuration"] = []interface{}{
			loginMode,
		}
	}
	return userAccount
}
```

Reply:
Thanks for pointing this out, I will follow this in future commitments.